### ISSUE TITLE
Auto adjust tree width if icons from toolbar and node name overlap

### ISSUE THREAD

**Issue by MaaaxiKing** [godotengine/godot-proposals#2034]
### Describe the project you are working on
Tycoon Game
### Describe the problem or limitation you are having in your project
I could not read the name of a node.
### Describe the feature / enhancement and how it helps to overcome the problem or limitation
The tree would automatically adjust its width, so the node name would be readable.
### Describe how your proposal will work, with code, pseudo-code, mock-ups, and/or diagrams
![Godot_Auto_Adjust_Tree_proposal](https://user-images.githubusercontent.com/60623532/103175258-e7f00e80-4868-11eb-88bb-84e865fa343b.png)
### If this enhancement will not be used often, can it be worked around with a few lines of script?
First, it will be used often. The automatical adjustment can not be worked around since it is core but manually, one could pull the window bigger.
### Is there a reason why this should be core and not an add-on in the asset library?
Yes, it's easy to implement and useful for everyone.

**Comment by Calinou** [comment]
The issue is that the left dock's width can only be increased so much until it overflows the editor window. This is especially true at small window sizes such as the ones you'd use on a 1366×768 display.

**Comment by MaaaxiKing** [comment]
Of course, it may be disturbing, so maybe it could be changed in the Editor Settings.

**Comment by Jummit** [comment]
Isn't this addressed by godotengine/godot#34779?

**Comment by MaaaxiKing** [comment]
I also thought about that but it is not the same. I think, the person who wrote "Better horizontal scroll for Tree" would find it good if the tree automatically adjusted itself but if you decrease the width, there should appear a horizontal scroll bar.

**Comment by dalexeev** [comment]
Just make the dock wider and the editor will remember that. However, for each project separately.
